### DISCUSSION TITLE
Read pixel value from image, darken the color, pass it to bordercolor

### DISCUSSION THREAD

**Discussion by smerfmaruda** [ImageMagick/ImageMagick#2480] [Help] [answered]
I want to add a border to the image, color of which should be based--but directly equal (like: darker or having its hue shifted--to value of a specific pixel.

Something like: `-set option:bcolor "%%[pixel:u.p{6,30}]" <modify bcolor> -bordercolor "%%[bcolor]"`.

I could, of course, generate a canvas of `bcolor`, darken / modify it, read its pixel value, set `bordercolor` correspondingly but that seems to a bit too complex for such a simple task.

Any hints, please?

**Comment by snibgo** [comment, marked as answer]
You can set a bordercolor from the colour of one of the pixels, as you suggest (Windows BAT syntax):
```
magick toes.png -set option:bcolor "%%[pixel:u.p{6,30}]" -bordercolor %%[bcolor] -border 2 x.png
```

But if you want to use a modified version of that colour, then I suggest you use IM's normal image processing. So you need an image of 1x1 pixels, and the obvious way to create that is with a crop of a clone. For example, we crop a pixel, and darken it by multiplying by 0.6:
```
magick ^
  toes.png ^
  ( +clone ^
    -crop 1x1+6+30 +repage ^
    -evaluate Multiply 0.6 ^
    -set option:bcolor "%%[pixel:u.p{0,0}]" ^
    +delete ^
  ) ^
  -bordercolor %%[bcolor] -border 2 ^
  x2.png
```

**Reply by smerfmaruda** [reply]
Got it. Kudos!
